py/desitarget/brightstar.py

```diff
@@ -0,0 +1,458 @@
+    #ADM to determine the truly correct numbers for DESI
+    radius = (0.0802*mags*mags - 1.860*mags + 11.625)
+
+    #ADM calculate the TARGETID
+    targetid = objs['BRICKID'].astype(np.int64)*1000000 + objs['OBJID']
+
+    #ADM create an output recarray that is just RA, Dec, TARGETID and the radius
+    done = objs[['RA','DEC']].copy()
+    done = rfn.append_fields(done,["TARGETID","RADIUS"],[targetid,radius],usemask=False,dtypes=['>i8','<f4'])
+
+    if outfilename is not None:
+        fitsio.write(outfilename, done, clobber=True)
+
+    return done
+
+
+def is_in_bright_star(targs,starmask):
+    """Determine whether a set of targets is in a bright star mask
+
+    Parameters
+    ----------
+    targs : :class:`recarray`
+        A recarray of targets as made by desitarget.cuts.select_targets
+    starmask : :class:`recarray`
+        A recarray containing a bright star mask as made by desitarget.brightstar.make_bright_star_mask
+
+    Returns
+    -------
+        mask : array_like. True if target is in a bright star mask  
+    """
+
+    #ADM initialize an array of all False (nothing is yet in a star mask)
+    done = np.zeros(len(targs), dtype=bool)
+
+    #ADM turn the coordinates of the masks and the targets into SkyCoord objects
+    ctargs = SkyCoord(targs["RA"]*u.degree, targs["DEC"]*u.degree)
+    cstars = SkyCoord(starmask["RA"]*u.degree, starmask["DEC"]*u.degree)
+    
+    #ADM this is the largest search radius we should need to consider
+    #ADM in the future an obvious speed up is to split on radius 
+    #ADM as large radii are rarer but take longer
+    maxrad = max(starmask["RADIUS"])*u.arcmin
+
+    #ADM coordinate match the star masks and the targets
+    idtargs, idstars, d2d, d3d = cstars.search_around_sky(ctargs,maxrad)
+
+    #ADM catch the case where nothing fell in a mask
+    if len(idstars) == 0:
+        return done
+
+    #ADM for a matching star mask, find the angular separations that are less than the mask radius
+    w = np.where(d2d.arcmin < starmask[idstars]["RADIUS"])
+
+    #ADM any matching idtargs that meet this separation criterion are in a mask (at least one)
+    done[idtargs[w]] = 'True'
+
+    return done
+
+
+def set_target_bits(targs,starmask):
+    """Apply bright star mask to targets, return desi_target array
+
+    Parameters
+    ----------
+    targs : :class:`recarray`
+        A recarray of targets as made by desitarget.cuts.select_targets
+    starmask : :class:`recarray`
+        A recarray containing a bright star mask as made by desitarget.brightstar.make_bright_star_mask
+
+    Returns
+    -------
+        an ndarray of the updated desi_target bit that includes bright star information
+                                                                                                                                                      
+    To Do
+    -----
+        Currently sets IN_BRIGHT_OBJECT but should also match on the TARGETID to set BRIGHT_OBJECT bit
```

Comment:
Should also set `NEAR_BRIGHT_OBJECT` for a larger radius, or add that to the TODO list.